View transition layers are positioned wrong if a clipping layer is present.

mattwoodrow opened the pull request:
#### c76f59cd157b8728cdac41d7336553dc84c3297f
<pre>
View transition layers are positioned wrong if a clipping layer is present.
<a href="https://bugs.webkit.org/show_bug.cgi?id=270058">https://bugs.webkit.org/show_bug.cgi?id=270058</a>
&lt;<a href="https://rdar.apple.com/123607249">rdar://123607249</a>&gt;

Reviewed by Tim Nguyen.

Positioning for a view-transition &apos;new&apos; layer currently sets the origin of the
primary graphics rect to 0,0, but this doesn&apos;t move any clipping layers positioned
in updateClippingStackLayerGeometry.

We can instead adjust the position of the &apos;parent graphics layer rect&apos; (since all
local layer positions are computed relative to that) to a value that results in 0,0
being computed for the primary rect.

* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateGeometry):

Canonical link: <a href="https://commits.webkit.org/275307@main">https://commits.webkit.org/275307@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8df8290cd2a721816f868340707ec39a5d40a410

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41476 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20490 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43854 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44043 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37568 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43783 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23596 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17820 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34291 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42050 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17417 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35719 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14951 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15133 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36722 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45403 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37663 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37042 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40796 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16291 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13368 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39200 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17910 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9297 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17965 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17554 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
This looks OK, but can we have a test?

mattwoodrow committed:
View transition layers are positioned wrong if a clipping layer is present.
https://bugs.webkit.org/show_bug.cgi?id=270058
<rdar://123607249>

Reviewed by Tim Nguyen.

Positioning for a view-transition 'new' layer currently sets the origin of the
primary graphics rect to 0,0, but this doesn't move any clipping layers positioned
in updateClippingStackLayerGeometry.

We can instead adjust the position of the 'parent graphics layer rect' (since all
local layer positions are computed relative to that) to a value that results in 0,0
being computed for the primary rect.

* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateGeometry):

Canonical link: https://commits.webkit.org/275307@main
